app/helpers/application_helper/button/transform_vm_button.rb

```diff
@@ -0,0 +1,12 @@
+class ApplicationHelper::Button::TransformVmButton < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def visible?
+    @record.vendor == "vmware"
+  end
+
+  def disabled?
+    # Is there a provider that supports import?
+    EmsInfra.all.select(&:validate_import_vm).empty?
+  end
```

Comment:
on its own `validate_import_vm` a simple version comparison (for RHV) or stubbed to return always `false`. With this in mind it is IMHO just a question of how expensive is to query all infra providers - do users usually have 100s or 1000s of them?